sdk/python/kfp/components_tests/test_components.py

```diff
@@ -693,6 +699,64 @@ def test_check_task_object_no_output_attribute_when_multiple_outputs(self):
 
         self.assertFalse(hasattr(task, 'output'))
 
+    def test_edit_component_loaded_from_text(self):
```

Comment:
Can we try to add a test case where we actually mutate the spec of the component and assert the changes is respected? Thanks

Reply:
+1, I am interested to see how to programmatically edit a component, and how does it benefit users.

Reply:
Programmatic editing is already possible:

```python
my_op = load_component_from_uri(...)

# Modify my_op.component_spec
my_op.component_spec.save("new.component.yaml")
```

This is not intended for programmatic editing, but rather for users of Jupyter notebooks: You call .edit() and the editable component spec is added as the next cell.